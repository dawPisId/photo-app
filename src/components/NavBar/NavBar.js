import { Container, Header, Logo } from "./NavBar.styled.js";

import AccountCircleIcon from "@material-ui/icons/AccountCircle";
import NavComponent from "../NavComponent/NavComponent.js";
import PhotoAlbumIcon from "@material-ui/icons/PhotoAlbum";
import PhotoIcon from "@material-ui/icons/Photo";
import React from "react";

const navData = [
  {
    label: "PHOTOS",
    link: "/",
    exact: true,
    icon: <PhotoIcon />,
  },
  {
    label: "ALBUMS",
    link: "/albums",
    exact: false,
    icon: <PhotoAlbumIcon />,
  },
  {
    label: "PROFILE",
    link: "/profile/1",
    exact: false,
    icon: <AccountCircleIcon />,
  },
];

const NavBar = () => (
  <Header>
    <Container>
      <Logo>PHOTOS APP</Logo>
      {navData.map((data, index) => (
        <NavComponent
          link={data.link}
          label={data.label}
          isExact={data.exact}
          icon={data.icon}
          key={index}
        />
      ))}
    </Container>
  </Header>
);

export default NavBar;
