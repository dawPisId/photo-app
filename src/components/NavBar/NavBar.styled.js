import styled from "styled-components";

export const Header = styled.div`
  display: block;
`;

export const Container = styled.div`
  padding: 0rem 2rem;
  margin: 0;
  display: flex;
  height: 4rem;
  background-color: white;
  box-shadow: 0px 0px 10px grey;
  font-family: "Roboto" sans-serif;
  justify-content: flex-end;
  align-items: center;
  list-style-type: none;
  a {
    text-decoration: none;
    color: black;
  }
  @media (max-width: 576px) {
    height: auto;
    flex-direction: column;
    a {
      padding: 1rem;
    }
  }
`;

export const Item = styled.li`
  font-size: 1.125rem;
  margin: 0rem 0.5rem;
  svg {
    vertical-align: sub;
    padding: 0 0.5rem;
  }
  span {
    display: inline-block;
  }
  @media (max-width: 576px) {
    margin: 0;
    padding: 2rem;
  }
`;

export const Logo = styled.li`
  font-family: "Roboto", sans-serif;
  font-size: 1.25rem;
  margin-right: auto;
  @media (max-width: 576px) {
    margin: 1rem 0rem 1rem 0rem;
  }
`;
