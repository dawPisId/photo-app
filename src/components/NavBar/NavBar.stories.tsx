import { ComponentMeta, ComponentStory } from "@storybook/react";

import { BrowserRouter } from "react-router-dom";
import NavBar from "./NavBar";
import React from "react";

export default {
  title: "Photo App/NavBar",
  component: NavBar,
} as ComponentMeta<typeof NavBar>;

const Template: ComponentStory<typeof NavBar> = (args) => (
  <BrowserRouter>
    <NavBar />
  </BrowserRouter>
);

export const Main = Template.bind({});
Main.args = {};
