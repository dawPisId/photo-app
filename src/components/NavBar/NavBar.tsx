import { Container, Header, Logo } from "./NavBar.styled";
import React, { FC } from "react";

import AccountCircleIcon from "@material-ui/icons/AccountCircle";
import NavItem from "../NavItem/NavItem";
import PhotoAlbumIcon from "@material-ui/icons/PhotoAlbum";
import PhotoIcon from "@material-ui/icons/Photo";

const navData = [
  {
    label: "PHOTOS",
    link: "/",
    exact: true,
    icon: <PhotoIcon />,
  },
  {
    label: "ALBUMS",
    link: "/albums",
    exact: false,
    icon: <PhotoAlbumIcon />,
  },
  {
    label: "PROFILE",
    link: "/profile/1",
    exact: false,
    icon: <AccountCircleIcon />,
  },
];
/**
 * Navigation bar component used as a header.
 */
const NavBar: FC = () => (
  <Header>
    <Container>
      <Logo>PHOTOS APP</Logo>
      {navData.map(({ link, label, exact, icon }, index) => (
        <NavItem
          link={link}
          label={label}
          isExact={exact}
          icon={icon}
          key={index}
        />
      ))}
    </Container>
  </Header>
);

export default NavBar;
