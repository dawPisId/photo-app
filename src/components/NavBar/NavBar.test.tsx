import { BrowserRouter } from "react-router-dom";
import NavBar from "./NavBar";
import React from "react";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
