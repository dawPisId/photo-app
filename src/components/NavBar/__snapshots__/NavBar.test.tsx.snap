// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdnxRM drXKXV"
>
  <div
    className="sc-gtsrHT eQRIqZ"
  >
    <li
      className="sc-hKFxyN eQpwhn"
    >
      PHOTOS APP
    </li>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <li
        className="sc-eCApnc hpMDYO"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 19V5c0-1.1-.9-2-2-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2zM8.5 13.5l2.5 3.01L14.5 12l4.5 6H5l3.5-4.5z"
          />
        </svg>
        PHOTOS
      </li>
    </a>
    <a
      aria-current={null}
      href="/albums"
      onClick={[Function]}
    >
      <li
        className="sc-eCApnc hpMDYO"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 2H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zM6 4h5v8l-2.5-1.5L6 12V4zm0 15l3-3.86 2.14 2.58 3-3.86L18 19H6z"
          />
        </svg>
        ALBUMS
      </li>
    </a>
    <a
      aria-current={null}
      href="/profile/1"
      onClick={[Function]}
    >
      <li
        className="sc-eCApnc hpMDYO"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
          />
        </svg>
        PROFILE
      </li>
    </a>
  </div>
</div>
`;
